text/0000-contingency-plan.md

```diff
@@ -0,0 +1,425 @@
+- Feature Name: N/A
+- Start Date: 2015-05-07
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+This RFC has two main goals:
+
+- define what precisely constitutes a breaking change for the Rust language itself;
+- define a language versioning mechanism that extends the sorts of
+  changes we can make without causing compilation failures (for
+  example, adding new keywords).
+  
+# Motivation
+
+With the release of 1.0, we need to establish clear policy on what
+precisely constitutes a "minor" vs "major" change to the Rust language
+itself (as opposed to libraries, which are covered by [RFC 1105]).
+**This RFC proposes limiting breaking changes to changes with
+soundness implications**: this includes both bug fixes in the compiler
+itself, as well as changes to the type system or RFCs that are
+necessary to close flaws uncovered later.
+
+However, simply landing all breaking changes immediately could be very
+disruptive to the ecosystem. Therefore, **the RFC also proposes
+specific measures to mitigate the impact of breaking changes**, and
+some criteria when those measures might be appropriate.
+
+Furthermore, there are other kinds of changes that we may want to make
+which feel like they *ought* to be possible, but which are in fact
+breaking changes. The simplest example is adding a new keyword to the
+language -- despite being a purely additive change, a new keyword can
+of course conflict with existing identifiers. Therefore, **the RFC
+proposes a simple annotation that allows crates to designate the
+version of the language they were written for**. This effectively
+permits some amount of breaking changes by making them "opt-in"
+through the version attribute.
+
+However, even though the version attribute can be used to make
+breaking changes "opt-in" (and hence not really breaking), this is
+still a tool to be used with great caution. Therefore, **the RFC also
+proposes guidelines on when it is appropriate to include an "opt-in"
+breaking change and when it is not**.
```

Comment:
@seanmonstar 

> So, eventually npm stopped treating this as an error, and simply a warning.

Interesting, thanks! I *think* I wrote that the equivalent scenario (specifying a version of Rust newer than the current compiler) would be a warning here as well...